src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -73,113 +75,103 @@ namespace Impl {
-    nnz_lno_type            nr;          // num_rows  (# verts)
-    nnz_lno_type            nc;          // num cols
-    size_type               ne;          // # edges
-    const_lno_row_view_type xadj;        // rowmap, transpose of rowmap
-    const_lno_nnz_view_t    adj;         // entries, transpose of entries   (size = # edges)
-    const_clno_row_view_t   t_xadj;      // rowmap, transpose of rowmap
-    const_clno_nnz_view_t   t_adj;       // entries, transpose of entries
-
+    lno_t     nv;          // num vertices
+    lno_t     nr;          // num_rows  (# verts)
+    lno_t     nc;          // num cols
+    size_type ne;          // # edges
+    rowmap_t  xadj;        // rowmap, transpose of rowmap
+    entries_t adj;         // entries, transpose of entries   (size = # edges)
+    rowmap_t  t_xadj;      // transpose rowmap (aliases xadj if !doing_bipartite)
+    entries_t t_adj;       // transpose entries (aliases adj if !doing_bipartite)
     HandleType* gc_handle;      // pointer to the graph coloring handle
 
-
   private:
     int  _chunkSize;      // the size of the minimum work unit assigned to threads.  Changes the convergence on GPUs
     int  _max_num_iterations;
-    char _use_color_set;      // The VB Algorithm Type: 0: VB,  1: VBCS,  2: VBBIT  (1, 2 not implemented).
     bool _ticToc;             // if true print info in each step
     bool _verbose;            // if true, print verbose information
-
+    bool using_edge_filtering;
+    static constexpr color_type UNPROCESSED = 0;
+    static constexpr color_type UNCOLORABLE = ~UNPROCESSED;
+    static constexpr color_type CONFLICTED = UNCOLORABLE - 1;
 
   public:
     /**
      * \brief GraphColorDistance2 constructor.
      * \param nv_: number of vertices in the graph
-     * \param ne_: number of edges in the graph
      * \param row_map: the xadj array of the graph. Its size is nv_ +1
-     * \param entries: adjacency array of the graph. Its size is ne_
+     * \param entries: adjacency array of the graph.
      * \param handle: GraphColoringHandle object that holds the specification about the graph coloring,
      *    including parameters.
      */
-    GraphColorDistance2(nnz_lno_type            nv_,
-                        nnz_lno_type            nc_,
-                        size_type               ne_,
-                        const_lno_row_view_type row_map,
-                        const_lno_nnz_view_t    entries,
-                        const_clno_row_view_t   t_row_map,
-                        const_clno_nnz_view_t   t_entries,
-                        HandleType*             handle)
+    GraphColorDistance2(lno_t       nv_,
+                        lno_t       nc_,
+                        rowmap_t    row_map,
+                        entries_t   entries,
+                        rowmap_t    t_row_map,
+                        entries_t   t_entries,
+                        HandleType* handle)
         : nv(nv_)
         , nr(nv_)
```

Comment:
Is this always true or is this an assumption?
If the latter it should be tested... for a prolongator matrix the number of columns will be small compared to the number of rows.

Reply:
@lucbv Rows and vertices are always equivalent (I should just get rid of 'nv'). You couldn't do distance-2 coloring on a nonsquare prolongator matrix, but you could do bipartite row or column coloring.